Neos.ContentRepository/Classes/Domain/Service/PublishingService.php

```diff
@@ -184,6 +184,9 @@ protected function doDiscardNode(NodeInterface $node, array &$alreadyDiscardedNo
                 $parentBasePath = $node->getPath();
                 $affectedChildNodeDataInSameWorkspace = $this->nodeDataRepository->findByParentAndNodeType($parentBasePath, null, $node->getWorkspace(), null, false, true);
```

Comment:
I wonder why you get nodes from other workspaces? As we pass the workspace here, I would expect just nodes of the given workspace, right? Maybe we should rather check `findByParentAndNodeType` if something is wrong, than fixing the result afterward. WDYT?

Reply:
Hmm, ok. It's because "live" is a baseworkspace of your workspace:
https://github.com/neos/neos-development-collection/blob/e480aaae2fa7a309b917535d16cfe2bc01901e7b/Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php#L592-L592

And for added nodes and removed nodes we filter afterwards:
https://github.com/neos/neos-development-collection/blob/e480aaae2fa7a309b917535d16cfe2bc01901e7b/Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php#L543-L543
https://github.com/neos/neos-development-collection/blob/e480aaae2fa7a309b917535d16cfe2bc01901e7b/Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php#L567

And if we would do that for found nodes too, we would disable the "shine through" of the base workspace for other usage of `getNodeDataForParentAndNodeType`.

This is actually tricky.

Reply:
Yes this is the problem, and I'm not sure why the issue appears just now and how to 100% reproduce this yet.
But I can't find any indication why it worked before, as the idea of `findByParentAndNodeType` is that it finds the nodes from the base workspace too, as you say.